emscripten.py

```diff
@@ -708,15 +708,14 @@ def proxy_debug_print(call_type):
 
 def include_asm_consts(pre, forwarded_json, metadata):
   if shared.Settings.WASM and shared.Settings.SIDE_MODULE:
-    assert len(metadata['asmConsts']) == 0, 'EM_ASM is not yet supported in shared wasm module (it cannot be stored in the wasm itself, need some solution)'
+    if metadata['asmConsts']:
+      exit_with_error('EM_ASM is not yet supported in shared wasm module (it cannot be stored in the wasm itself, need some solution)')
 
   asm_consts, all_sigs, call_types = all_asm_consts(metadata)
   asm_const_funcs = []
-  for s in range(len(all_sigs)):
-    sig = all_sigs[s]
+  for sig, call_type in zip(all_sigs, call_types):
```

Comment:
This isn't quite equivalent, if all_sigs is longer it skips the last few sigs.
Not sure when that happens though. Probably worth an assert that they're equal-length if that's something we can assume?